src/parser/vp9_parser.cpp

```diff
@@ -208,14 +301,17 @@ ParserResult Vp9VideoParser::SendPicForDecode() {
     p_pic_param->profile = p_uncomp_header->profile;
     p_pic_param->bit_depth = p_uncomp_header->color_config.bit_depth;
 
-    // Todo
-    for (int i = 0; i < VP9_NUM_REF_FRAMES; i++)
-        p_pic_param->reference_frames[i] = 0; 
-
+    for (int i = 0; i < VP9_NUM_REF_FRAMES; i++) {
+        if (dpb_buffer_.virtual_buffer_index[i] != INVALID_INDEX) {
+            p_pic_param->reference_frames[i] = dpb_buffer_.frame_store[dpb_buffer_.virtual_buffer_index[i]].dec_buf_idx;
+        } else {
+            p_pic_param->reference_frames[i] = 0xFF;
```

Comment:
You mean memset the reference_frames array to 0xFF first? But this is 32-bit array. If we pre-set using a loop, it will be slower than the current code.

Reply:
That is not necessarily true. Conditional jump statements are usually costly. We can get rid of else condition by setting the default vale

Reply:
Suggested change added.